api/src/main/java/org/openmrs/ServiceOrder.java

```diff
@@ -0,0 +1,184 @@
+	/**
+	 * @return the specimenSource
+	 */
+	public Concept getSpecimenSource() {
+		return specimenSource;
+	}
+
+	/**
+	 * @param specimenSource the specimenSource to set
+	 */
+	public void setSpecimenSource(Concept specimenSource) {
+		this.specimenSource = specimenSource;
+	}
+
+	/**
+	 * Gets the laterality.
+	 *
+	 * @return the laterality.
+	 */
+	public Laterality getLaterality() {
+		return laterality;
+	}
+
+	/**
+	 * Sets the laterality.
+	 *
+	 * @param laterality the laterality to set.
+	 */
+	public void setLaterality(ServiceOrder.Laterality laterality) {
+		this.laterality = laterality;
+	}
+
+	/**
+	 * Gets the clinical history.
+	 *
+	 * @return the clinical history.
+	 */
+	public String getClinicalHistory() {
+		return clinicalHistory;
+	}
+
+	/**
+	 * Sets the clinical history.
+	 *
+	 * @param clinicalHistory the clinical history to set.
+	 */
+	public void setClinicalHistory(String clinicalHistory) {
+		this.clinicalHistory = clinicalHistory;
+	}
+
+	/**
+	 * Gets frequency of a particular order
+	 */
+	public OrderFrequency getFrequency() {
+		return frequency;
+	}
+
+	/**
+	 * Sets frequency of a particular order
+	 *
+	 * @param frequency
+	 */
+	public void setFrequency(OrderFrequency frequency) {
+		this.frequency = frequency;
+	}
+
+	/**
+	 * Gets numberOfRepeats of a particular order
+	 */
+	public Integer getNumberOfRepeats() {
+		return numberOfRepeats;
+	}
+
+	/**
+	 * Sets numberOfRepeats of a particular order
+	 *
+	 * @param numberOfRepeats to set
+	 */
+	public void setNumberOfRepeats(Integer numberOfRepeats) {this.numberOfRepeats = numberOfRepeats;}
+
+	/**
+	 * Gets a location of the particular order
+	 */
+	public Location getLocation() {return location;}
+
+	/**
+	 * Sets a location of the particular order
+	 * @param location to set
+	 */
+	public void setLocation(Location location) {this.location = location;}
+
+	/**
+	 * Creates a discontinuation order for this.
+	 *
+	 * @see org.openmrs.Order#cloneForDiscontinuing()
+	 * @return the newly created order
+	 * <strong>Should</strong> set all the relevant fields
+	 */
+	@Override
+	public Order cloneForDiscontinuing() {
+		Order newOrder = new Order();
+		newOrder.setCareSetting(getCareSetting());
+		newOrder.setConcept(getConcept());
+		newOrder.setAction(Order.Action.DISCONTINUE);
+		newOrder.setPreviousOrder(this);
+		newOrder.setPatient(getPatient());
+		newOrder.setOrderType(getOrderType());
+		return newOrder;
+	}
+
+	/**
+	 * Creates a particular order for revision from this order, sets the previousOrder, action field and
+	 * other particular order fields.
+	 *
+	 * @return the newly created order
+	 * <strong>Should</strong> set all the relevant fields
+	 * <strong>Should</strong> set the relevant fields for a DC order
+	 */
+	@Override
+	public Order cloneForRevision() {
+		return cloneForRevisionHelper(new Order());
+	}
+
+	/**
+	 * @see Order#cloneForRevisionHelper(Order)
+	 */
+	protected abstract Order cloneForRevisionHelper(Order target);
```

Comment:
Did you intentionally leave the above method empty?

Reply:
Yes , I was thinking we could carry out the implementation in the child classes ,due to the conflicts encountered.

Reply:
Which conflicts were those?